scipy/spatial/_geometric_slerp.py

```diff
@@ -185,7 +185,7 @@ def geometric_slerp(start,
                          "space")
 
     if np.array_equal(start, end):
-        return [start] * np.asarray(t).size
+        return np.linspace(start, start, np.asarray(t).size)
```

Comment:
I pushed in this simplified patch--I suggest that if CI remains happy we merge this and then deal with some of the other discussion points in follow-up(s).

Maybe this will also help move the original typing PR forward with the correct return type, even if the shape issues are still confusing.